test/functional/outputs/azuremonitor/forward_to_azuremonitor_test.go

```diff
@@ -0,0 +1,141 @@
+	"strings"
+	"time"
+
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/test/helpers/azuremonitor"
+	"github.com/openshift/cluster-logging-operator/test/helpers/rand"
+	v1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/framework/functional"
+)
+
+var _ = Describe("Forwarding to Azure Monitor Log ", func() {
+	const azureSecret = "azure-secret"
+	var (
+		framework  *functional.CollectorFunctionalFramework
+		mockServer *azuremonitor.MockServer
+		secret     *v1.Secret
+		sharedKey  = rand.Word(16)
+		inputName  = "custom-app"
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewCollectorFunctionalFrameworkUsingCollector(logging.LogCollectionTypeVector)
+		// Start mockServer server
+		mockServer = azuremonitor.NewMockServer(framework.Namespace, azuremonitor.Mockoon)
+		Expect(mockServer.Create(framework.Test.Client)).To(Succeed())
+
+		framework.Forwarder.Spec.Inputs = append(framework.Forwarder.Spec.Inputs,
+			logging.InputSpec{
+				Name:        inputName,
+				Application: &logging.Application{},
+			},
+		)
+
+		// Set up the common template forwarder configuration.
+		// HTTP Data Collector API URI	has format
+		// https://<CustomerId>.<Host>/api/logs?api-version=2016-04-01
+		// <CustomerID>	The unique identifier for the Log Analytics workspace.
+		// <Host> alternative host for dedicated Azure regions.
+		// We will split route host by dot (.) and use first part as customer id, second part as alternative host
+		// this allow as emulate forwarding
+		host := mockServer.Route.Spec.Host
+		dot := strings.Index(host, ".")
+		customerId := host[:dot]
+		altHost := host[dot+1:]
+		framework.Forwarder.Spec.Outputs = append(framework.Forwarder.Spec.Outputs,
+			logging.OutputSpec{
+				Name: logging.OutputTypeAzureMonitor,
+				Type: logging.OutputTypeAzureMonitor,
+				OutputTypeSpec: logging.OutputTypeSpec{
+					AzureMonitor: &logging.AzureMonitor{
+						LogType:    "myLogType",
+						CustomerId: customerId,
+						Host:       altHost,
+					},
+				},
+				TLS: &logging.OutputTLSSpec{
+					InsecureSkipVerify: true,
+				},
+				Secret: &logging.OutputSecretSpec{
+					Name: azureSecret,
+				},
+			})
+
+		framework.Forwarder.Spec.Pipelines = append(framework.Forwarder.Spec.Pipelines,
+			logging.PipelineSpec{
+				InputRefs:  []string{inputName},
+				OutputRefs: []string{logging.OutputTypeAzureMonitor},
+				Name:       "pip-pip",
+			},
+		)
+
+		secret = runtime.NewSecret(framework.Namespace, azureSecret,
+			map[string][]byte{
+				constants.SharedKey: sharedKey,
+			},
+		)
+
+		framework.Secrets = append(framework.Secrets, secret)
+	})
+
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	It("should accept application logs", func() {
```

Comment:
Is there no place for us to read the received logs from the mock to check them in stead of only checking the collector logs?

Reply:
no, we can only check logs on mock server side with rules and send back corresponding response

Reply:
What about file logging? https://mockoon.com/docs/latest/logging-and-recording/requests-logging/#file-logging

Reply:
Nice idea, i will try. I didn't think in this way. Thanks

Reply:
done